### DISCUSSION TITLE
how to aggregate different API responses to create one response (append, or create a response json) and send that back as a response to one API call?

### DISCUSSION THREAD

**Discussion by Ronit-at-coalition** [Kong/kong#11369] [Help] [answered]
how to aggregate different API responses to create one response (append, or create a response json) and send that back as a response to one API call?

**Comment by hanshuebner** [comment, marked as answer]
@Ronit-at-coalition You will need to write a custom plugin for this.  From the plugin, you can make multiple http requests using [lua-resty-http](https://github.com/ledgetech/lua-resty-http) and combine the responses as required.  If you need your upstream requests go through the complete Kong request logic, you can send requests back to Kong Gateway in a loopback fashion.

Hope this helps,
Hans
